### PR TITLE
Add OnClosed event for WebTransportClientInterface.

### PR DESCRIPTION
**Pull request by jianjunz**

### CODE REVIEW

File: web_transport/sdk/impl/tests/web_transport_echo_visitors.cc [hunk 1]
```diff
@@ -46,11 +46,11 @@ ServerEchoVisitor::~ServerEchoVisitor() {}
 
 void ServerEchoVisitor::OnSession(WebTransportSessionInterface* session) {
   CHECK(session);
-  LOG(INFO) << "Server on session.";
   std::unique_ptr<SessionEchoVisitor> visitor =
       std::make_unique<SessionEchoVisitor>();
   session->SetVisitor(visitor.get());
   session_visitors_.push_back(std::move(visitor));
+  sessions_.push_back(session);
```
Comment (taste1981) [inline web_transport/sdk/impl/tests/web_transport_echo_visitors.cc:53]: session_visitors_.emplace_back(visitor);  // might look better?

File: web_transport/sdk/impl/tests/web_transport_echo_visitors.h [hunk 2]
```diff
@@ -49,8 +49,13 @@ class ServerEchoVisitor : public WebTransportServerInterface::Visitor {
   void OnEnded() override {}
   void OnSession(WebTransportSessionInterface* session) override;
 
+  std::vector<WebTransportSessionInterface*> Sessions() { return sessions_; }
```
Comment (taste1981) [inline web_transport/sdk/impl/tests/web_transport_echo_visitors.h:52]: const

File: web_transport/sdk/impl/web_transport_owt_client_impl.cc [hunk 3]
```diff
@@ -150,8 +150,13 @@ void WebTransportOwtClientImpl::OnError(const net::WebTransportError& error) {
 }
 
 void WebTransportOwtClientImpl::OnClosed(
-      const absl::optional<net::WebTransportCloseInfo>& close_info) {
-  LOG(INFO) << "OnClosed.";
+    const absl::optional<net::WebTransportCloseInfo>& close_info) {
+  if (!visitor_) {
```
Comment (taste1981) [inline web_transport/sdk/impl/web_transport_owt_client_impl.cc:154]: one common comment. consider adding calling thread checker for those callbacks?

File: web_transport/sdk/impl/web_transport_owt_client_impl.cc [hunk 4]
```diff
@@ -150,8 +150,13 @@ void WebTransportOwtClientImpl::OnError(const net::WebTransportError& error) {
...
+    return;
+  }
+  visitor_->OnClosed(
+      close_info.has_value() ? close_info->code : 0,
```
Comment (taste1981) [inline web_transport/sdk/impl/web_transport_owt_client_impl.cc:158]: you should avoid calling close_info.has_value() twice.

File: web_transport/sdk/impl/web_transport_owt_client_impl.cc (hunk 3 above)
Comment (taste1981) [reply web_transport/sdk/impl/web_transport_owt_client_impl.cc:154]: no need for the brackets here.

File: web_transport/sdk/impl/tests/web_transport_echo_visitors.cc (hunk 1 above)
Comment (jianjunz) [reply web_transport/sdk/impl/tests/web_transport_echo_visitors.cc]: Replaced with emplace_back.

File: web_transport/sdk/impl/tests/web_transport_echo_visitors.h (hunk 2 above)
Comment (jianjunz) [reply web_transport/sdk/impl/tests/web_transport_echo_visitors.h]: Done.

File: web_transport/sdk/impl/web_transport_owt_client_impl.cc (hunk 3 above)
Comment (jianjunz) [reply web_transport/sdk/impl/web_transport_owt_client_impl.cc]: This method is called by `net::WebTransportClient`. It's supposed to be called on correct thread. We can add a thread checker for these methods after threading model change (described in PR #60).

File: web_transport/sdk/impl/web_transport_owt_client_impl.cc (hunk 4 above)
Comment (jianjunz) [reply web_transport/sdk/impl/web_transport_owt_client_impl.cc]: Done.

**Review by taste1981 — APPROVED** [review]

**Pull request merged**